### PR TITLE
MSI - RHODS / GPU tests

### PR DESCRIPTION
**Pull request by rnetser**

### CODE REVIEW

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-operators-v4.13-rhods-gpu-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-operator-v4.13-rhods-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-operators-v4.13-rhods-gpu-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-operators-v4.13-rhods-gpu-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

File: ci-operator/step-registry/addon/install/addon-install-commands.sh [hunk 1]
```diff
@@ -7,15 +7,27 @@ set -o verbose
 
 CLUSTER_NAME=$(cat "${SHARED_DIR}/cluster-name")
 OCM_TOKEN=$(cat /var/run/secrets/ci.openshift.io/cluster-profile/ocm-token)
```
Comment (omrirh) [inline ci-operator/step-registry/addon/install/addon-install-commands.sh:9]: we should add `BREW_TOKEN` here as well

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

File: ci-operator/step-registry/addon/install/addon-install-commands.sh (hunk 1 above)
Comment (rnetser) [reply ci-operator/step-registry/addon/install/addon-install-commands.sh:9]: done

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-operators-v4.13-rhods-gpu-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-servicemesh-operators-v4.13-rhods-gpu-servicemesh-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-operator-v4.13-rhods-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-operator-v4.13-rhods-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-operator-v4.13-rhods-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-servicemesh-operators-v4.13-rhods-gpu-servicemesh-tests

**Review by myakove — APPROVED** [review]

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-servicemesh-operators-v4.13-rhods-gpu-servicemesh-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-servicemesh-operators-v4.13-rhods-gpu-servicemesh-tests

File: ci-operator/step-registry/addon/install/addon-install-commands.sh [hunk 2]
```diff
@@ -7,15 +7,29 @@ set -o verbose
...
+    --api-host ${API_HOST} \
+    --timeout ${TIMEOUT} \
+    --brew-token ${BREW_TOKEN} \
+    "
+
+ADDONS_CMD=""
+for i in {1..4}; do
+  if [[ -n $(eval "echo $(eval "echo $"ADDON$i"_CONFIG")") ]]; then
+    CMD=$(eval "echo $(eval "echo $"ADDON$i"_CONFIG")")
+    ADDONS_CMD="${ADDONS_CMD} --addons ${CMD}"
```
Comment (omrirh) [inline ci-operator/step-registry/addon/install/addon-install-commands.sh:23]:
```suggestion
  if [[ -n "$(eval "echo \$ADDON${i}_CONFIG")" ]]; then
    CMD="$(eval "echo \$ADDON${i}_CONFIG")"
    ADDONS_CMD+=" --addons \"${CMD}\""
```

File: ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh [hunk 3]
```diff
@@ -7,15 +7,28 @@ set -o verbose
...
+    --token ${OCM_TOKEN} \
+    --api-host ${API_HOST} \
+    --timeout ${TIMEOUT} \
+    "
+
+ADDONS_CMD=""
+for i in {1..4}; do
+  if [[ -n $(eval "echo $(eval "echo $"ADDON$i"_CONFIG")") ]]; then
+    CMD=$(eval "echo $(eval "echo $"ADDON$i"_CONFIG")")
+    ADDONS_CMD="${ADDONS_CMD} --addons ${CMD}"
```
Comment (omrirh) [inline ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh:21]:
```suggestion
  if [[ -n "$(eval "echo \$ADDON${i}_CONFIG")" ]]; then
    CMD="$(eval "echo \$ADDON${i}_CONFIG")"
    ADDONS_CMD+=" --addons \"${CMD}\""
```

File: ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh [hunk 4]
```diff
@@ -7,15 +7,28 @@ set -o verbose
...
+  fi
+done
+
+RUN_COMMAND="${RUN_COMMAND} ${ADDONS_CMD}"
+
+if [ -n "${PARALLEL}" ]; then
```
Comment (omrirh) [inline ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh:27]: ditto

File: ci-operator/step-registry/addon/install/addon-install-commands.sh (hunk 2 above)
Comment (rnetser) [reply ci-operator/step-registry/addon/install/addon-install-commands.sh:23]: logic was changed

File: ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh (hunk 3 above)
Comment (rnetser) [reply ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh]: see updated logic

File: ci-operator/step-registry/addon/install/addon-install-commands.sh [hunk 5]
```diff
@@ -7,15 +7,29 @@ set -o verbose
...
+  fi
+done
+
+RUN_COMMAND="${RUN_COMMAND} ${ADDONS_CMD}"
+
+if [ -n "${PARALLEL}" ]; then
```
[the remark this answers is not in the document]
Comment (rnetser) [reply ci-operator/step-registry/addon/install/addon-install-commands.sh]: no need to have the 1st check, the 2nd condition will not be met if the param is not set

File: ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh (hunk 4 above)
Comment (rnetser) [reply ci-operator/step-registry/addon/uninstall/addon-uninstall-commands.sh]: ditto

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-gpu-servicemesh-operators-v4.13-rhods-gpu-servicemesh-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-operator-v4.13-rhods-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by rnetser** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by myakove** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by myakove** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rdbaas-operator-addon-v4.12-poc-tests

**Comment by myakove** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-operator-v4.13-rhods-tests

**Pull request merged**